### PR TITLE
Transferred from another project

### PR DESCRIPTION
**Pull request by scalarindetail**
This PR is the whole implementation.
We would like to transfer it from an internal project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,146 @@
...
+
+This is a client-side library for web applications to interact with [Scalar DL](https://github.com/scalar-labs/scalardl) network.
+
+## Install
+
+We can use package managers to install this library.
+
+**NPM**
+```
+npm install @scalar-labs/scalardl-web-sdk
```
Comment (feeblefakie) [inline README.md]: Let's name it `scalardl-web-client-sdk`.
Please update the related ones also.
Sorry for saying this here.

File: README.md [hunk 2]
```diff
@@ -1 +1,146 @@
...
+```
+
+**Yarn**
+```
+yarn add @scalar-labs/scalardl-web-sdk
+```
+
+Or, download the bundle file from.
```
[the remark this answers is not in the document]
Comment (scalarindetail) [reply README.md]: Sorry, I forgot to remove this.
I was thinking to finish the URL after having the release address.
We can do it later.

File: README.md (hunk 1 above)
Comment (scalarindetail) [reply README.md]: Understood.

**Comment by scalarindetail** [comment]
We rename related name to `scalardl-web-client-sdk` and the NPM package is named `@scalar-labs/scalardl-web-client-sdk`.

Please kindly review again.

File: example/index.html [hunk 3]
```diff
@@ -0,0 +1,252 @@
+<html>
+
+<head>
+    <script src="../dist/scalardl-web-client-sdk.bundle.js"></script>
+</head>
+
+<body>
+    <h3>証明書の登録</h3>
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
    <h3>Register a certificate</h3>
```

File: example/index.html [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        <div>
+            <button id="registerCertificate">
+                RegisterCertificate
+            </button>
+            <span id="registerCertificateResult"></span>
+        </div>
+    </div>
+
+    <hr>
+    <h3>コントラクトの登録</h3>
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
    <h3>Register a contract</h3>
```

File: example/index.html [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        <div>
+            <button id="registerContract">
+                RegisterContract
+            </button>
+            <span id="registerContractResult"></span>
+        </div>
+    </div>
+
+    <hr>
+    <h3>登録したコントラクトの一覧</h3>
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
    <h3>List registered contracts</h3>
```

File: example/index.html [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+    <div>
+        <button id="listContracts">
+            ListContracts
+        </button>
+        <span id="listContractsResult"></span>
+    </div>
+
+    <hr>
+    <h3>コントラクトの実行</h3>
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
    <h3>Execute a contract</h3>
```

File: example/index.html [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        <div>
+            <button id="executeContract">
+                ExecuteContract
+            </button>
+            <span id="executeContractResult"></span>
+        </div>
+    </div>
+
+    <hr>
+    <h3>アセットを検証</h3>
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
    <h3>Validate an asset in the ledger</h3>
```

File: example/index.html [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+            'scalar.ledger.client.cert_holder_id': holderId,
+            'scalar.ledger.client.private_key_pem': key,
+            'scalar.ledger.client.cert_pem': certificate,
+            'scalar.ledger.client.cert_version': 1,
+            'scalar.ledger.client.tls.enabled': tls 
+        };
+        clientService = new Scalar.ClientService(clientProperties);
+        clientService.registerCertificate().then(response => {
+            if (response.getStatus() == Scalar.StatusCode.OK) {
+                document.querySelector("#registerCertificateResult").innerHTML = "<b>証明書登録成功</b>";
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
                document.querySelector("#registerCertificateResult").innerHTML = "<b>Success</b>";
```

File: example/index.html [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+            return;
+        }
+        if (!clientService) {
+            window.alert("You need to register a certificate berfore register any contracts");
+            return;
+        }
+        clientService.registerContract(contractId, contractName, contractBinary, contractProperties).then(
+            response => {
+                if (response.getStatus() == Scalar.StatusCode.OK) {
+                    document.querySelector("#registerContractResult").innerHTML = "<b>コントラクト登録成功</b>";
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
                    document.querySelector("#registerContractResult").innerHTML = "<b>Success</b>";
```

File: example/index.html [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+            return;
+        }
+        const argument = JSON.parse(argumentJSON);
+        if (!clientService) {
+            window.alert("You need to register a certificate berfore execution");
+            return;
+        }
+        clientService.executeContract(executeContractId, argument).then(response => {
+            if (response.getStatus() == Scalar.StatusCode.OK) {
+                document.querySelector("#executeContractResult").innerHTML = "<b>コントラク実行成功</b><br>";
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
                document.querySelector("#executeContractResult").innerHTML = "<b>Success</b><br>";
```

File: example/index.html [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+            return;
+        }
+        if (!clientService) {
+            window.alert("You need to register a certificate berfore execution");
+            return;
+        }
+        clientService.validateLedger(asset).then(response => {
+
+            if (response.getStatus() == Scalar.StatusCode.OK) {
+                document.querySelector("#validateLedgerResult").innerHTML = "アセットは改ざんされていない";
```
Comment (feeblefakie) [inline example/index.html]:
```suggestion
                document.querySelector("#validateLedgerResult").innerHTML = "The specified asset is not tampered";
```

**Pull request merged**